MapboxCoreNavigation/UpcomingRouteAlertInfo.swift

```diff
@@ -0,0 +1,84 @@
+import Foundation
+import CoreLocation
+import MapboxNavigationNative
+import MapboxDirections
+
+/**
+ `UpcomingRouteAlertInfo` encapsulates information about various incoming events. Common attributes like location, distance to the event, length and other is provided for each POI, while specific meta data is supplied via `alert` property.
+ */
+public struct UpcomingRouteAlertInfo {
+    public enum RouteAlertType {
+        case incident(Incident)
+        case tunnel(TunnelInfo)
+        case borderCrossing(BorderCrossingInfo)
+        case tollCollection(TollCollection)
+        case serviceArea(RestStop)
+    }
```

Comment:
This enumeration indicates more than just the type, so we should rename it to just `RouteAlert`. There will probably be a conflict with MapboxNavigationNative, but we can solve that conflict by selective imports and namespaces. In the worst-case scenario, a separate file could import MapboxNavigationNative and declare `InternalRouteAlert` as a type alias of `MapboxNavigationNative.RouteAlert`.